### PR TITLE
datepicker: memory leak

### PR DESCRIPTION
**Pull request by jigar140291**
Ticket: [15250](https://bugs.jqueryui.com/ticket/15270)

### CODE REVIEW

**Comment by fnagel** [comment]
Hey @jigar140291 thanks for your contribution and sorry for the delay.

Sounds reasonable to me and I'm able to debug this, but not within memory debugger functionality. Can you help me out here?

@arschmitz What do you think about this?

**Comment by fnagel** [comment]
@jigar140291 Would you mind adding a test for this?

**Comment by jigar140291** [comment]
@fnagel Sorry for the delay. I will look into the test cases.

**Comment by jigar140291** [comment]
@fnagel I have added test cases. could you please review it.

File: tests/unit/datepicker/methods.js [hunk 1]
```diff
@@ -8,9 +8,18 @@ define( [
...
 	var inl,
-		inp = testHelper.init( "#inp" );
+		inp = testHelper.init( "#inp" ),
+		dp = $( "#ui-datepicker-div" );
+
+	// Destroy and clear active reference
+	inp.datepicker( "show" );
+	assert.equal( dp.css( "display" ), "block", "Datepicker - visible" );
+	inp.datepicker( "hide" ).datepicker( "destroy" );
+	assert.ok( !$.datepicker._curInst, "Datepicker - destroyed and cleared reference" );
```
Comment (mgol) [inline tests/unit/datepicker/methods.js]: Could you make it more specific here?
```suggestion
	assert.ok( $.datepicker._curInst == null, "Datepicker - destroyed and cleared reference" );
```

Comment (jigar140291) [reply tests/unit/datepicker/methods.js]: @mgol that make sense. Thanks !!

**Comment by jigar140291** [comment]
@mgol unit test case updated.

**Comment by jigar140291** [comment]
@fnagel it's done. could you please review and merge.

**Comment by fnagel** [comment]
@jigar140291 Sorry but now there are even more commits. Wanna try again or should I do this for you?

**Comment by jigar140291** [comment]
@fnagel seems like I squash it incorrectly. Could you do that for me. Thanks!!

**Comment by jigar140291** [comment]
@fnagel I have squashed commits into one single commit. please check and merge.

**Comment by jigar140291** [comment]
@arschmitz Could you please review and merge it.

**Pull request merged**